presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -152,6 +157,8 @@ public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, Sym
         private final boolean preferStreamingOperators;
         private final boolean redistributeWrites;
         private final boolean scaleWriters;
+        private final String partitioningProvider;
```

Comment:
`partitioningProviderCatalogName`

Reply:
Maybe just `partitioningProviderCatalog` to be consistent with the property?